arch/risc-v/src/mpfs/mpfs_sdio.c

```diff
@@ -0,0 +1,179 @@
+ *   slotno - Not used.
+ *
+ * Returned Values:
+ *   A reference to an SDIO interface structure.  NULL is returned on
+ *   failures.
+ *
+ ****************************************************************************/
+
+struct sdio_dev_s; /* See include/nuttx/sdio.h */
+struct sdio_dev_s *sdio_initialize(int slotno)
+{
+  switch (slotno)
+  {
+  case 0:
+#ifdef CONFIG_MPFS_EMMCSD
+    return mpfs_emmcsd_sdio_initialize(slotno);
+#endif
+
+  case 1:
+#ifdef CONFIG_MPFS_COREMMC
+    return mpfs_coremmc_sdio_initialize(slotno);
+#endif
+
+  default:
+    break;
+  }
+
+  mcerr("sdio slot number %d not supported!\n", slotno);
+  return NULL;
+}
+
+/****************************************************************************
+ * Name: sdio_mediachange
+ *
+ * Description:
+ *   Called by board-specific logic -- possible from an interrupt handler --
+ *   in order to signal to the driver that a card has been inserted or
+ *   removed from the slot
+ *
+ * Input Parameters:
+ *   dev        - An instance of the SDIO driver device state structure.
+ *   cardinslot - true is a card has been detected in the slot; false if a
+ *                card has been removed from the slot.  Only transitions
+ *                (inserted->removed or removed->inserted should be reported)
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void sdio_mediachange(struct sdio_dev_s *dev, bool cardinslot)
+{
+  struct mpfs_dev_s *priv = (struct mpfs_dev_s *)dev;
+
+  if (!dev)
+    {
+      mcerr("sdio device not found\n");
+      return;
+    }
+
+  switch (priv->hw_base)
+  {
+#ifdef CONFIG_MPFS_EMMCSD
+  case MPFS_EMMC_SD_BASE:
+    mpfs_emmcsd_sdio_mediachange(dev, cardinslot);
+    return;
+#endif
+
+#ifdef CONFIG_MPFS_COREMMC
+  case CONFIG_MPFS_COREMMC_BASE:
+    mpfs_coremmc_sdio_mediachange(dev, cardinslot);
+    return;
+#endif
+
+  default:
+    break;
+  }
+
+  mcerr("Invalid sdio base address\n");
+}
+
+/****************************************************************************
+ * Name: sdio_wrprotect
+ *
+ * Description:
+ *   Called by board-specific logic to report if the card in the slot is
+ *   mechanically write protected.
+ *
+ * Input Parameters:
+ *   dev       - An instance of the SDIO driver device state structure.
+ *   wrprotect - true is a card is writeprotected.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void sdio_wrprotect(struct sdio_dev_s *dev, bool wrprotect)
+{
+  struct mpfs_dev_s *priv = (struct mpfs_dev_s *)dev;
+
+  if (!dev)
+    {
+      mcerr("sdio device not found\n");
+      return;
+    }
+
+  switch (priv->hw_base)
+  {
+#ifdef CONFIG_MPFS_EMMCSD
+  case MPFS_EMMC_SD_BASE:
+
+    mpfs_emmcsd_sdio_wrprotect(dev, wrprotect);
+    return;
+#endif
+
+#ifdef CONFIG_MPFS_COREMMC
+  case CONFIG_MPFS_COREMMC_BASE:
+    mpfs_coremmc_sdio_wrprotect(dev, wrprotect);
+    return;
+#endif
```

Comment:
Please keep consistent styling, and adhere to the nuttx coding standard here.

Reply:
No sure why the checkpatch does not claim about those switch case styles..
Fixed.

Reply:
Hmm, CI might have the format check disabled /missing due to my mistake. I'll check this ASAP.

Reply:
Nope, the format check works, strange that it did not catch this error.

Reply:
Yes, I ran a test and it does seem like the format check does not notice formatting errors with switch-case statements